tests/components/roborock/test_sensor.py

```diff
@@ -47,3 +53,49 @@ async def test_sensors(hass: HomeAssistant, setup_entry: MockConfigEntry) -> Non
         hass.states.get("sensor.roborock_s7_maxv_last_clean_end").state
         == "2023-01-01T03:43:58+00:00"
     )
+
+
+async def test_listener_update(
+    hass: HomeAssistant, setup_entry: MockConfigEntry
+) -> None:
+    """Test that when we receive a mqtt topic, we successfully update the entity."""
+    assert hass.states.get("sensor.roborock_s7_maxv_status").state == "charging"
+    # Listeners are global based on uuid - so this is okay
+    client = RoborockMqttClient(
+        USER_DATA, DeviceData(device=HomeDataDevice("abc123", "", "", "", ""), model="")
+    )
+    # Test Status
+    with patch(
+        "roborock.protocol._Parser.parse",
+        return_value=(
+            [
+                RoborockMessage(
+                    protocol=RoborockMessageProtocol.GENERAL_REQUEST,
+                    payload=b'{"t": 1699464794, "dps": {"121": 5}}',
+                )
+            ],
+            b"",
+        ),
+    ), patch("roborock.api.AttributeCache.value", STATUS.as_dict()):
+        # Symbolizes a mqtt message coming in
+        client.on_message(None, None, AsyncMock())
```

Comment:
Does it work to call `on_message_received` to avoid the message parser and just pass in the `RoborockMessage` directly here?

Reply:
Yes that works - there was a problem that made me not originally do that - but I must have solved the problem because it worked perfect this time. Thanks appreciate the review - this looks a lot saner now